TAGradingServer/account/account-numerictext-gradeable.php

```diff
@@ -0,0 +1,410 @@
+                                </td>
+HTML;
+            $question_grades=pgArrayToPhp($temp['grade_value_array']);
+            //return an empty array of zeros here
+            if (empty($question_grades)) {
+                $question_grades = array();
+                for ($i = 0; $i < $num_numeric; ++$i) {
+                    $question_grades[$i] = 0;
+                }
+            }
+            // calculate the overall grade for this nt_gradeable
+            $total_grade = array_sum($question_grades);
+            $text_fields = pgArrayToPhp($temp['grade_text']);
+           
+            if (empty($text_fields)) {
+                $text_fields = array();
+                for ($i = 0; $i < $num_text; ++$i) {
+                    $text_fields[$i] = "";
+                }
+            }
+            
+            for($i = 0; $i < $num_numeric; ++$i) {
+                print <<<HTML
+                                <td class="input-container" style="border: 1px solid black">
+                                    <input id="cell-{$nt_row["g_id"]}-{$row["user_id"]}-q{$i}" type="text" value="{$question_grades[$i]}" />
+                                </td>
+HTML;
+            }
+            print <<<HTML
+                                <td style="width: 10px" id="cell-{$nt_row["g_id"]}-{$row['user_id']}-score">{$total_grade}</td>
+HTML;
+            
+            //print the text fields
+            for ($i = $num_numeric; $i <$num_numeric+$num_text; ++$i) {
+                $text_field = isset($text_fields[$i]) ? $text_fields[$i] : "";
+                print <<<HTML
+                                <td class="input-container" style="border: 1px solid black">
+                                    <input id="cell-{$nt_row["g_id"]}-{$row["user_id"]}-t{$i}" elem="text" type="text" value="{$text_field}" />
+                                </td>
+HTML;
+            }
+            print <<<HTML
+                            </tr>
+HTML;
+        }
+    }
+	print <<<HTML
+                        </tbody>
+                    </table>
+                </div>
+HTML;
+    $first = false;
+}
+print <<<HTML
+            </div>
+        </div>
+    </div>
+</div>
+HTML;
+
+$js_array_questions = "";
+$js_array_text = "";
+foreach ($nt_gradeables as $nt_gradeable) {
+    $params = array($nt_gradeable['g_id']);
+    $db->query("SELECT COUNT(*) AS cnt from gradeable AS g INNER JOIN gradeable_component AS gc ON g.g_id=gc.g_id WHERE g.g_id=? AND gc_is_text='false'", $params);
+    $num_numeric = $db->row()['cnt'];
+    $db->query("SELECT COUNT(*) AS cnt from gradeable AS g INNER JOIN gradeable_component AS gc ON g.g_id=gc.g_id WHERE g.g_id=? AND gc_is_text='true'", $params);
+    $num_text = $db->row()['cnt'];
+    $js_array_questions .= $nt_gradeable['g_id'].':"'.$num_numeric.'",';
+    $js_array_text .= $nt_gradeable['g_id'].':"'.$num_text.'",';
+}
+
+echo <<<HTML
+	<script type="text/javascript">
```

Comment:
The indentation is reflective of the HTML being printed, not other PHP code. As for the part specifically pointed out in this comment, is the objection that SQL queries should be formatted on separate lines?